vcd/resource_vcd_catalog_item.go

```diff
@@ -236,3 +244,65 @@ func createOrUpdateCatalogItemMetadata(d *schema.ResourceData, meta interface{})
 	}
 	return nil
 }
+
+// Imports a CatalogItem into Terraform state
+// This function task is to get the data from vCD and fill the resource data container
+// Expects the d.ID() to be a path to the resource made of Org name + dot + Catalog name
+//
+// Example import path (id): org_name.catalog_name.catalog_item_name
+func resourceVcdCatalogItemImport(d *schema.ResourceData, meta interface{}) ([]*schema.ResourceData, error) {
+	resourceURI := strings.Split(d.Id(), ".")
+	if len(resourceURI) != 3 {
+		return nil, fmt.Errorf("resource name must be specified as org.catalog.catalog_item")
+	}
+	orgName, catalogName, catalogItemName := resourceURI[0], resourceURI[1], resourceURI[2]
+
+	if orgName == "" {
+		return nil, fmt.Errorf("import: empty org name provided")
+	}
+	if catalogName == "" {
+		return nil, fmt.Errorf("import: empty catalog name provided")
+	}
+	if catalogItemName == "" {
+		return nil, fmt.Errorf("import: empty catalog item name provided")
+	}
+
+	vcdClient := meta.(*VCDClient)
+	adminOrg, err := vcdClient.GetAdminOrgByName(orgName)
+	if err != nil {
+		return nil, fmt.Errorf(errorRetrievingOrg, orgName)
+	}
+
+	catalog, err := adminOrg.GetCatalogByName(catalogName, false)
+	if err != nil {
+		return nil, govcd.ErrorEntityNotFound
+	}
+
+	catalogItem, err := catalog.GetCatalogItemByName(catalogItemName, false)
+	if err != nil {
+		return nil, govcd.ErrorEntityNotFound
+	}
+
+	err = d.Set("org", orgName)
```

Comment:
More of a team discussion. I guess we could agree what we do with `d.Set()` pattern everywhere.

Personally I don't think it is worth putting an `err` check everywhere in it as it makes the code quite huge in an unlikely (if ever possible) place for `err`. Quite a few providers ignore it. I'd vote to ignore it with `_ = d.Set` to avoid annoying linters.

Reply:
I agree. We should be consistent. I will adopt the `_ = d.Set` format

Reply:
I even vote for cleaner approach:)
`d.Set`

Reply:
`d.Set` alone is not cleaner, as it causes IDEs and linters to show warnings

Reply:
Yeah. I agree with dataclouder, even though I used (d.Set()) myself. I guess we should put the _ = d.Set... to avoid linting errors in all major editors.

Reply:
I changed all the `err = d.Set` into `_ = d.Set`